### DISCUSSION TITLE
Improving Chainsaw Test with ytt Templating

### DISCUSSION THREAD

**Discussion by shubham-cmyk** [kyverno/chainsaw#733] [Ideas]
Hi, 
I have looked at the tool: https://github.com/carvel-dev/ytt
This provides an approach for the yaml template since the chainsaw accepts yaml we can use this library to provide the template solution. 

I want to write some doc/example tests. 
How we can simplify the writing/updating/handling of the Chainsaw Test. 

CC @eddycharly WDYT ?

**Comment by eddycharly** [comment]
TBH i'm not a big fan of ytt at all.

**Reply by shubham-cmyk** [reply]
Do you have some other ideas ?

**Reply by eddycharly** [reply]
PKL looks nice https://pkl-lang.org/index.html
